src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests_CollectionExpression.cs

```diff
@@ -0,0 +1,1772 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.UseCollectionInitializer;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseCollectionInitializer;
+
+using VerifyCS = CSharpCodeFixVerifier<
+    CSharpUseCollectionInitializerDiagnosticAnalyzer,
+    CSharpUseCollectionInitializerCodeFixProvider>;
+
+[Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)]
+public partial class UseCollectionInitializerTests_CollectionExpression
+{
+    private static async Task TestInRegularAndScriptAsync(string testCode, string fixedCode, OutputKind outputKind = OutputKind.DynamicallyLinkedLibrary)
+    {
+        await new VerifyCS.Test
+        {
+            ReferenceAssemblies = Testing.ReferenceAssemblies.NetCore.NetCoreApp31,
+            TestCode = testCode,
+            FixedCode = fixedCode,
+            LanguageVersion = LanguageVersion.Preview,
+            TestState = { OutputKind = outputKind }
+        }.RunAsync();
+    }
+
+    private static Task TestMissingInRegularAndScriptAsync(string testCode)
+        => TestInRegularAndScriptAsync(testCode, testCode);
+
+    [Fact]
+    public async Task TestNotOnVarVariableDeclarator()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    var c = [|new|] List<int>();
+                    [|c.Add(|]1);
+                }
+            }
+            """,
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    var c = new List<int>
+                    {
+                        1
+                    };
+                }
+            }
+            """);
```

Comment:
some tests show that we will *not* produce collection expressions, but will still produce collection-initializers.